main.go

```diff
@@ -315,9 +315,22 @@ func main() {
 			cacheHandler = wrapIdleHandler(cacheHandler, idleTimer, accessLogger, httpServer)
 		}
 
-		mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-			metricsMdlw.Handler(r.Method, http.HandlerFunc(cacheHandler)).ServeHTTP(w, r)
-		})
+		if c.EnableEndpointMetrics {
+			metricsMdlw := httpmiddleware.New(httpmiddleware.Config{
+				Recorder: httpmetrics.NewRecorder(httpmetrics.Config{
+					DurationBuckets: durationBuckets,
+				}),
+			})
+			mux.Handle("/metrics", metricsMdlw.Handler("metrics", promhttp.Handler()))
+			mux.Handle("/status", metricsMdlw.Handler("status", http.HandlerFunc(h.StatusPageHandler)))
+			mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+				metricsMdlw.Handler(r.Method, http.HandlerFunc(cacheHandler)).ServeHTTP(w, r)
+			})
+		} else {
+			mux.Handle("/metrics", promhttp.Handler())
```

Comment:
Should this be skipped if metrics are disabled? Or maybe replaced by a 404 handler?

Reply:
good point, removed !